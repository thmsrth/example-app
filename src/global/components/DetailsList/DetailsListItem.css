.DetailsListItem__title {
    padding: var(--section-padding);
    vertical-align: top;
}

.DetailsListItem__value {
    padding: var(--section-padding);
    vertical-align: top;
    text-align: right;
}
