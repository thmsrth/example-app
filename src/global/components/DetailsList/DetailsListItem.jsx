import React from 'react';

import PropTypes from 'prop-types';
import keyType from './types/key';

import { bemCls } from '../../utils/ClassNameHelpers';

import DetailsListItemValue from './DetailsListItemValue';

import './DetailsListItem.css';

const DetailsListItem = ({ keyProp, className, ...otherProps }) => (
  <tr
    className={`DetailsListItem ${className}`}
    tabIndex="0"
  >
    <td className={`${bemCls(`DetailsListItem ${className}`, '__title')}`}>
      {keyProp.title}
    </td>
    <td className={`${bemCls(`DetailsListItem ${className}`, '__value')}`}>
      <DetailsListItemValue
        {...otherProps}
        className={bemCls(className, 'Value')}
        keyProp={keyProp}
      />
    </td>
  </tr>
);

export default DetailsListItem;

DetailsListItem.propTypes = {
  className: PropTypes.string,
  keyProp: keyType,
};

DetailsListItem.defaultProps = {
  className: '',
  keyProp: {},
};
