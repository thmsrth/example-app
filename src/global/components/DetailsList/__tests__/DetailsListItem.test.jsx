import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import DetailsListItem from '../DetailsListItem';

const props = {
  value: 'huhu',
  keyProp: {
    name: 'description', title: 'Description:', type: 'text',
  },
};

describe('DetailsListItem', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<DetailsListItem
      {...props}
    />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
