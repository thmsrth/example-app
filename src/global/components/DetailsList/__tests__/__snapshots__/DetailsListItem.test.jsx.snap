// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsListItem should render the component correctly 1`] = `
<tr
  className="DetailsListItem "
  tabIndex="0"
>
  <td
    className="DetailsListItem__title __title"
  >
    Description:
  </td>
  <td
    className="DetailsListItem__value __value"
  >
    <DetailsListItemValue
      className="Value"
      keyProp={
        Object {
          "name": "description",
          "title": "Description:",
          "type": "text",
        }
      }
      value="huhu"
    />
  </td>
</tr>
`;
